integration_test/ops_agent_test.go

```diff
@@ -2461,29 +2458,54 @@ func runResourceDetectorCli(ctx context.Context, logger *logging.DirectoryLogger
 			remote: "go.sum"},
 	}
 
+	// Create the folder structure on the VM
 	workDir := path.Join(workDirForPlatform(vm.Platform), "run_resource_detector")
+	packageDir := path.Join(workDir, "confgenerator", "resourcedetector")
+	var createFolderCmd string
+	if gce.IsWindows(vm.Platform) {
+		createFolderCmd = fmt.Sprintf("New-Item -ItemType Directory -Path %s", strings.ReplaceAll(packageDir, "/", "\\"))
+	} else {
+		createFolderCmd = fmt.Sprintf("mkdir -p %s", packageDir)
+	}
+	out, err := gce.RunScriptRemotely(ctx, logger, vm, createFolderCmd, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create folder with %s in VM: %s", createFolderCmd, out.Stderr)
+	}
+
+	// Upload the files
 	for _, file := range filesToUpload {
 		f, err := os.Open(file.local)
 		if err != nil {
 			return nil, err
 		}
-		err = gce.UploadContent(ctx, logger, vm, f, path.Join(workDir, file.remote))
+		var remotePath string
+		if gce.IsWindows(vm.Platform) {
+			remotePath = strings.ReplaceAll(path.Join(workDir, file.remote), "/", "\\")
+		} else {
+			remotePath = path.Join(workDir, file.remote)
+		}
+		err = gce.UploadContent(ctx, logger, vm, f, remotePath)
```

Comment:
```suggestion
		remotePath := path.Join(workDir, file.remote)
		if gce.IsWindows(vm.Platform) {
			remotePath = strings.ReplaceAll(remotePath, "/", "\\")
		}
		err = gce.UploadContent(ctx, logger, vm, f, remotePath)
```

Reply:
Good suggestion! Together with the next suggestion, I have removed the different commands of Windows/Non-Windows and simplify them to the same command.